skyd/api.go

```diff
@@ -132,11 +140,16 @@ func (api *api) ResolveSkylink(skylink string) (string, error) {
 		}{}
 		err = json.NewDecoder(resp.Body).Decode(&resolved)
 		if err != nil {
-			return "", errors.AddContext(err, "bad response body from skyd")
+			return skylink, errors.AddContext(err, "bad response from skyd")
+		}
+		err = skylink.LoadString(resolved.Skylink)
+		if err != nil {
+			return skylink, errors.AddContext(err, "unable to decode resolved skylink")
 		}
-		return resolved.Skylink, nil
+		return skylink, nil
 	}
 
+	// in all other cases simply return the original skylink
```

Comment:
All other cases are errors right? So we should probably try to read the error message and return that instead.

Reply:
done